libs/common-transfers/src/lib/services/excluded-dates.service.ts

```diff
@@ -1,62 +1,37 @@
 import { Injectable } from '@angular/core';
-import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { AbstractControl, ValidationErrors, ValidatorFn, AsyncValidatorFn } from '@angular/forms';
 import { RestrictedDatesHttpService, RestrictedDates } from '../../../../transfers-dates-http-ang/src';
-import { take } from 'rxjs/operators';
+import { map, shareReplay } from 'rxjs/operators';
+import { Observable } from 'rxjs';
 
 @Injectable()
 export class ExcludedDatesService {
-  excludedDates = [];
-  startDate: string;
-  endDate: string;
+  restrictedDates$: Observable<RestrictedDates> = this.restrictedDatesHttpService.restrictedDatesGet().pipe(
+    shareReplay()
+  );
 
-  constructor(private restrictedDatesHttpService: RestrictedDatesHttpService) {
-    this.restrictedDatesHttpService.restrictedDatesGet().pipe(take(1)).subscribe((dates: RestrictedDates) => {
-      this.startDate = dates.startDate;
-      this.endDate = dates.endDate;
-      this.excludedDates = dates.restrictedDates;
-    });
-  }
+  startDate$ = this.restrictedDates$.pipe(map((dates: RestrictedDates) => dates.startDate));
+  endDate$ = this.restrictedDates$.pipe(map((dates: RestrictedDates) => dates.endDate));
+  isDateValid$: Observable<(date: string) => boolean> = this.restrictedDates$.pipe(
+    map((dates) => {
+      return (date: string) => dates.restrictedDates.includes(date);
+    })
+  )
 
-  isDateValid(date: string) {
-    return this.excludedDates.includes(date);
-  }
+  constructor(private restrictedDatesHttpService: RestrictedDatesHttpService) { }
 
-  excludedDateValidator(): ValidatorFn {
-    return (control: AbstractControl): ValidationErrors | null => {
+  excludedDateValidator(): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors> | null => {
       const selectedDate = new Date(control.value);
       let month = selectedDate.getUTCMonth() + 1;
       let day = selectedDate.getUTCDate();
       let year = selectedDate.getUTCFullYear();
-      return this.isDateValid(`${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`)
-        ? { excludedDate: { value: control.value } }
-        : null;
-    };
-  }
-
-  pastDateValidator(): ValidatorFn {
```

Comment:
Realized while debugging: we don't need these. The design system datepicker already applies validators for the min / max date properties OOB. It is not overwritten by our `setValidators` call. It sets the errors `dateLessThanMinDate` and `dateGreaterThanMaxDate`, respectively. What is missing, however, is error messages for does errors, so we still need to write them: see comment in the component file.